### ISSUE TITLE
Correct workflow to perform SCTransform v2 on separate or merged dataset

### ISSUE THREAD

**Issue by reneemoerkens** [satijalab/sctransform#161]
Hey,

Thanks for the great work!

I am struggling with finding the correct workflow for my dataset, with regards to SCTransform and performing it on my merged dataset or individual datasets as specified in the vignette (https://satijalab.org/seurat/articles/sctransform_v2_vignette.html).

A bit more detail about my experiment:
I have single-cell RNA sequencing data from intestinal cells: 5 **stimulation conditions** for 3 **cell lines** = 15 samples. These 15 samples were multiplexed in a randomized way over 5 **lanes** in the 10x Genomics chip, giving 5 libraries after preparation (each lane/library containing a multiplexed sample of the 3 cell lines with a randomized combination of stimulation conditions). The 5 libraries were equimolarly pooled for sequencing. 
All experiments and library preparation were performed at the same time by the same individuals.

In my data I see some sources of variation:
-	The sequencing depth varies per **lane** (as some of the lanes contained slightly more cells than other lanes). In the Umap, the differences between lanes is not very visible, they are nicely mixed.
-	The 3 **cell lines** fall on a slightly different position in the Umap, which I want to correct for to improve annotation of cell types.

In the Seurat SCTransform v2 vignette I see that the Seurat object is split based on stimulation (IFN, CTRL), the SCTransform is performed on the separate datasets and then an integration is performed since there are differences in the position of these stimulation conditions in the UMAP. Since I have three different (intended and unintended) sources of variation to account for (i.e. **lanes** in 10x chip, **cell lines**, **stimulation conditions**), I was wondering which of these would be the preferred workflow:

![image](https://user-images.githubusercontent.com/129292461/228539601-cc4b3c25-13bd-440d-8ae1-9abf9412a152.png)

Basically, should I perform the SCTransform v2 on separate datasets or the complete merged dataset? And if I should split, based on which source of variation should I split the data (and before or after reciprocal PCA analysis to correct for cell line)? 

Thank you very much for your help!

**Comment by reneemoerkens** [comment]
Thank you so much for your fast response @saketkc, that helps a lot. So to confirm, this is the workflow that you propose:

![image](https://user-images.githubusercontent.com/129292461/228580368-f51d0c3f-823d-4404-8d37-23136a11377d.png)

And after this I can use the Integrated assay for cell type annotation and  SCT assay to run DE analysis (FindMarkers)

**Comment by saketkc** [comment]
Yes, this is what I suggested.

**Comment by saketkc** [comment]
Since this is all same technology I wouldn't except it to make a big difference. `PrepSCTFindMarkers` is useful to account for seqeuncing depth difference if you are using the merge strategy. So you can run SCT v2 independently on each cell line, perform integration and then run PrepSCTFindMarkers to adjust for the fact that the corrected counts calculated by SCT were based on the fact that different samples had different median UMI (so we reestimate the corrected counts using the minimum value of median UMI across samples). Does this make sense? 

In short, the integration choice is something that is very context dependent. Irrespective of which workflow you choose, you can always run `PrepSCTFindMarkers` before running DE to make sure your merged dataset has its corrected counts caclulcated based on one sequencing depth (which is the minimum of the median UMI across datasets).

**Comment by reneemoerkens** [comment]
@saketkc, I understand, and indeed, after testing different workflows I see that the differences are very small. With the different workflows I get a very similar list and number of differentially expressed genes per cluster, with similar log2FC and p_val_adj. 

In the end I will do my reciprocal PCA integration based on 'cell line' to improve annotation, then subsequently perform SCT v2 on the complete dataset (which is a bit more convenient in my follow-up analysis). Apparently in my dataset, the SCTranform v2 does a good job correcting for the sequence depth differences between the lanes, without performing splitting/independent SCTransform v2/PrepSCTFindMarkers. Could you shortly explain the method integrated within the SCTransform v2 function to account for sequencing depth differences? 

Thank you for all your help, very much appreciated!

**Comment by saketkc** [comment]
I will refer you to the [v2 paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02584-9) for the method details.

**Comment by reneemoerkens** [comment]
Thank you!

**Comment by reneemoerkens** [comment]
Hi @saketkc, sorry to bother you again, one final question:
I am considering to perform on the complete dataset: SCTransform(Seurat_object, vst.flavor = "v2", scale.factor = 21672.5), in which 21672.5 is the lowest median UMI when comparing dataset splitted by lane or cell_line or condition. Is it correct to use scale.factor when performing SCTransform v2 on the complete dataset (instead of splitted), to correct for sequencing depth differences? Or can I better leave it out? 

Also, when using scale.factor argument instead of PrepSCTFindMarkers, I don't need to pass recorrect_umi=FALSE when running FindMarkers() on a subset of the data. What is the reason you have to use recorrect_umi=FALSE (after PrepSCTFindMarkers) when analyzing a subset, because in most subset comparisons the difference in sequence depth remains a factor of technical variation, right?

**Comment by saketkc** [comment]
Yes you can pass a custom `scale_factor` (Note it is [scale_factor](https://github.com/satijalab/sctransform/blob/24b4ce630a38e2c3f600c2cc6d6ef015a58b5a14/R/vst.R#L137) and not `scale.factor`). This uses a constant scale factor (instead of estimating the median UMI from the data which might be different across datasets you are integrating). Once you do this, you can pass `recorrect_umi=FALSE` when running `FindMarkers`.
